### ISSUE TITLE
Headset not detected by SteamVR

### ISSUE THREAD

**Issue by meekah7** [alvr-org/ALVR#2124]
## Description

I'm trying to set up ALVR on my arch system for the first time, but no headset gets recognized by SteamVR, and ALVR says it can't find SteamVR.

I saw that I should check whether I "have software and hardware encoders installed" but I am unable to find out which encoders I need, or how to check that. I tried checking `ffmpeg -encoders` but that gives me a list with 100+ entries and I have no idea what to look for. I'm guessing HEVC is needed since it is mentioned all over the documentation, but it is present in the list:

```
 V....D libx265              libx265 H.265 / HEVC (codec hevc)
 V....D hevc_amf             AMD AMF HEVC encoder (codec hevc)
 V....D hevc_nvenc           NVIDIA NVENC hevc encoder (codec hevc)
 V..... hevc_qsv             HEVC (Intel Quick Sync Video acceleration) (codec hevc)
 V..... hevc_v4l2m2m         V4L2 mem2mem HEVC encoder wrapper (codec hevc)
 V....D hevc_vaapi           H.265/HEVC (VAAPI) (codec hevc)
```

I did notice when trying to reinstall ALVR, that some ALVR window launched when I launched SteamVR, despite me uninstalling ALVR just before. I also noticed the following message when reinstalling: `alvr_vulkan_layer (lib) generated 36 warnings`

## General Troubleshooting
- [x] I carefully followed the instructions in the [README](https://github.com/alvr-org/ALVR/blob/master/README.md) and successfully completed the setup wizard
- [x] I read the ALVR Wikis [polygraphene's ALVR GitHub Wiki](https://github.com/polygraphene/ALVR/wiki) and [ALVR GitHub Wiki](https://github.com/alvr-org/ALVR/wiki)

## Environment

arch linux, 6.8.7.arch1-2 kernel, KDE on wayland.

### Hardware
**Note**: for Linux, an upload to the [`hw-probe`](https://linux-hardware.org/) database is preferred: `hw-probe -all -upload`

**CPU**: AMD 5600X

**GPU**: AMD RX 6700XT

**GPU Driver Version**: Mesa 24.0.6-arch1.2

**Audio**: pipewire

### Installation
**ALVR Version**:
20.8.1-2 (AUR package name alvr, installed using yay)

**ALVR Settings File**:

**SteamVR Version**:
2.5.4

**Install Type**:
- [x] Packaged (`exe`, `deb`, `rpm`, etc) (if yay counts as packaged)
- [ ] Portable (`zip`)
- [ ] Source

**OS Name and Version** (`winver` on Windows or `grep PRETTY_NAME /etc/os-release` on most Linux distributions):
Arch Linux

**Comment by KishanPRao** [comment]
I had the same issue, and I was able to detect the headset after switching to previous (2.4.4) version.

**Comment by devurandom** [comment]
Is this related to https://github.com/alvr-org/ALVR/issues/2115 ?

**Comment by meekah7** [comment]
> Is this related to #2115 ?

Not sure, but I don't even get to connect my headset so I think it's different

**Comment by meekah7** [comment]
@KishanPRao How do you switch versions? I can only find instructions for windows. I also found [the protonDB steam VR depots](https://steamdb.info/app/250820/depots/?branch=v2.4.4), but I have no idea which files I need exactly or where to put them.

**Comment by KishanPRao** [comment]
@meekah7 Right click SteamVR, Properties, Betas, under Beta Participation select **previous**.

**Comment by meekah7** [comment]
@KishanPRao I did look at that menu, but it did not occur to me that this is a viable solution. I didn't see version numbers so I dismissed it. Thanks for pointing out where I went wrong!

**Comment by DanaGoyette** [comment]
I found that selecting the previous version didn't help at all.

When I looked at the logs to see why my headset wasn't even detected, I found this:
`Skipping external driver /usr/lib64/alvr because it is not a directory`.

By looking around in `ls -la /proc/$(pidof vrserver)/root/` (which seems to actually show the process's root, not the system root), I was able to see that the `/usr/lib64/alvr` directory isn't there.
But I also found out that there's a path `/run/host` that exposes the host root filesystem.

By editing `~/.config/openvr/openvrpaths.vrpath` to mention `/run/host/usr/lib64/alvr` instead of `/usr/lib64/alvr`, I was able to get the headset detected.   I also had to tell ALVR launcher to stop altering that file, via the "Extra" settings.

With the driver actually loading, it likes to crash SteamVR, resulting in SteamVR disabling it.  But even when it's not crashing, it's instead giving a black view in the headset, despite the VR preview showing what the headset should be showing.

**Comment by Vixea** [comment]
The black view is possibly because of the pic we use with the Vulkan layer I hope that hint helps but tbh I'm just fucking done with SteamVR crap they keep putting on us

**Comment by devurandom** [comment]
@DanaGoyette What are you running Steam with? Looks almost like Flatpak.

* Note that there is `flatpak enter com.valvesoftware.Steam /bin/bash`.
* https://github.com/alvr-org/ALVR/wiki/Flatpak explains how to run ALVR in the Steam Flatpak.
  - You do not have to build the Flatpak extension yourself, but can download a release `com.valvesoftware.Steam.Utility.alvr.flatpak` matching the version you have installed on your Quest from https://github.com/alvr-org/ALVR/releases.
  - You'll still have to patch the path to `libalvr_vulkan_layer.so`, see https://github.com/alvr-org/ALVR/issues/2043.

**Comment by DanaGoyette** [comment]
I actually used Debian's "steam-installer" package, which seems to be different from all three of the other options (Valve's deb, the Flatpak, and the Snap).  It's not an app container like Snap or Flatpak, but Steam seems to have its *own* container (the "pressure-vessel" thing).  

That container seems rather janky in several ways: it's missing some QT libraries, and missing some XDG-related packages, and it failed to register handlers for the `vrmonitor` protocol.

**Comment by meekah7** [comment]
forgot to mark as completed, because it works for me.

**Comment by DanaGoyette** [comment]
Well, maybe it works for you, but it sure doesn't work for me!  And I don't have permission to reopen it.

**Comment by Vixea** [comment]
## Steamvr
> Please set in SteamVR options -> Commandline options this:
> `~/.steam/steam/steamapps/common/SteamVR/bin/vrmonitor.sh %command%`
> On deb package-installed Steam (possibly mint, debian):
> `~/.steam/debian-installation/steamapps/common/SteamVR/bin/vrmonitor.sh %command%` 
> Adjust path if you have external drive
> **Without this, steam may not start properly, have black screen on client and you may have issues with pipewire device missing**

**Comment by DanaGoyette** [comment]
I finally got it to work, thanks to that info, and thanks to removing it from where I'd installed it in `/usr/lib64/alvr` / `/usr/share/alvr` / etc. and just running it from a self-built release tar.gz.  

Seems like ALVR could use some sort of documentation or code tweak to ensure users don't run into the `/usr/lib64/alvr is not a directory` issue.

To get the desktop streaming to work, I had to specifically *not* override `WAYLAND_DISPLAY`, and I think I even had to outright uninstall Gnome before KDE would show the "select what to screenshare" window.

Now I just need to figure out why the latency is bad even though ALVR claims the frame rate is high.
